core/src/main/java/org/akaza/openclinica/web/filter/ApiSecurityFilter.java

```diff
@@ -116,12 +116,22 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
 
                             if (userType.equals(UserType.SYSTEM.getName())){
                                 String clientId = decodedToken.get("clientId").toString();
+                                HashMap<String, String> userAccountToCreate = null;
```

Comment:
I had to use HashMap because userService.createUser method was expecting a HashMap. I didn't want to refactor that method since it is called from several other places.
Anyway, I refactored that code now.